app/sidero-controller-manager/internal/ipxe/ipxe_server.go

```diff
@@ -219,12 +217,6 @@ func ipxeHandler(w http.ResponseWriter, r *http.Request) {
 
 		return
 	}
-
-	if !strings.HasPrefix(env.ObjectMeta.Name, "agent") {
```

Comment:
I feel this is tough, but we need to find a solution here:

* if Talos <= 0.13, then we won't have any conditions on ServerBinding/MetalMachine, so we should keep doing our "dump" PXE booted if reached out to the PXE server 
* if Talos >= 0.14, the conditions will be present and we can use those vs. what we have right now

I don't know how to implement that fully, but at least with Talos <= 0.13 I think this code will never mark it as PXE booted?

Reply:
On other hand, PXE booting all the time if the server is configured to boot from network first is not the end of the world.

It's fine and we can live with that. It will break upgrades if someone does an upgrade, and might make big confusion if Environment gets updated, while installed Talos version is different.

Reply:
What I was thinking about is the following:

* if Talos >= 0.14, and Talos boots, it should send events and we should have _some_ conditions set on the ServerBinding from events; in that case we should stop PXE booting when install succeeds
* if Talos <= 0.13, we would have no conditions set on ServerBinding, but we have this "weak" condition that the server reached out to the iPXE, and we can keep using that?

Reply:
yeah... I forgot about backward compatibility here...
I wonder if we can detect if Talos <= 0.13 is used in Sidero somehow.

Reply:
The only idea I have is the above ^^: judge by the presence of conditions, by the time it gets to boot once again, it should have _some_ conditions set.

Reply:
Ok, I think then I need to update event handler to set conditions earlier